contracts/src/v0.8/dev/automation/2_1/KeeperRegistrar2_1.sol

```diff
@@ -0,0 +1,490 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "../../../interfaces/LinkTokenInterface.sol";
+import "./interfaces/AutomationRegistryInterface2_1.sol";
+import "../../../interfaces/TypeAndVersionInterface.sol";
+import "../../../ConfirmedOwner.sol";
+import "../../../interfaces/ERC677ReceiverInterface.sol";
+
+/**
+ * @notice Contract to accept requests for upkeep registrations
+ * @dev There are 2 registration workflows in this contract
+ * Flow 1. auto approve OFF / manual registration - UI calls `register` function on this contract, this contract owner at a later time then manually
+ *  calls `approve` to register upkeep and emit events to inform UI and others interested.
+ * Flow 2. auto approve ON / real time registration - UI calls `register` function as before, which calls the `registerUpkeep` function directly on
+ *  keeper registry and then emits approved event to finish the flow automatically without manual intervention.
+ * The idea is to have same interface(functions,events) for UI or anyone using this contract irrespective of auto approve being enabled or not.
+ * they can just listen to `RegistrationRequested` & `RegistrationApproved` events and know the status on registrations.
+ */
+contract KeeperRegistrar2_1 is TypeAndVersionInterface, ConfirmedOwner, ERC677ReceiverInterface {
+  /**
+   * DISABLED: No auto approvals, all new upkeeps should be approved manually.
+   * ENABLED_SENDER_ALLOWLIST: Auto approvals for allowed senders subject to max allowed. Manual for rest.
+   * ENABLED_ALL: Auto approvals for all new upkeeps subject to max allowed.
+   */
+  enum AutoApproveType {
+    DISABLED,
+    ENABLED_SENDER_ALLOWLIST,
+    ENABLED_ALL
+  }
+
+  bytes4 private constant REGISTER_REQUEST_SELECTOR = this.register.selector;
+
+  mapping(bytes32 => PendingRequest) private s_pendingRequests;
+
+  LinkTokenInterface public immutable LINK;
+
+  /**
+   * @notice versions:
+   * - KeeperRegistrar 2.1.0: Update interface to work with registry v2.1
```

Comment:
can we try to not change the interface? It will come at a cost of additional function call needed from user, but worth a discussion with @wentzeld